modules/phase_field/src/ics/TricrystalTripleJunctionIC.C

```diff
@@ -0,0 +1,128 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "TricrystalTripleJunctionIC.h"
+#include "MooseRandom.h"
+#include "MooseMesh.h"
+#include "MathUtils.h"
+
+template<>
+InputParameters validParams<TricrystalTripleJunctionIC>()
+{
+  InputParameters params = validParams<InitialCondition>();
+  params.addClassDescription("Tricrystal with a triple junction");
+  params.addRequiredParam<unsigned int>("op_num","Number of grain order parameters");
+  params.addRequiredParam<unsigned int>("op_index", "Index for the current grain order parameter");
+  params.addParam<Real>("theta1", 135.0, "Angle of first grain at triple junction in degrees");
+  params.addParam<Real>("theta2", 135.0, "Angle of second grain at triple junction in degrees");
+  return params;
+}
+
+TricrystalTripleJunctionIC::TricrystalTripleJunctionIC(const InputParameters & parameters) :
+    InitialCondition(parameters),
+    _mesh(_fe_problem.mesh()),
+    _nl(_fe_problem.getNonlinearSystem()),
+    _op_num(getParam<unsigned int>("op_num")),
+    _op_index(getParam<unsigned int>("op_index")),
+    _theta1(getParam<Real>("theta1")),
+    _theta2(getParam<Real>("theta2"))
+{
+  if (_op_num != 3)
+    mooseError("Tricrystal ICs must have op_num = 3");
+
+  if (_theta1 + _theta2 >= 360.0)
+    mooseError("Sum of the angles must total less than 360 degrees");
+
+  // Convert the angles to radians
+  _theta1 = _theta1 * libMesh::pi / 180.0;
+  _theta2 = _theta2 * libMesh::pi / 180.0;
+
+  // Change the first angle to be measured from the +x axis
+  _theta1 = 3.0 * libMesh::pi / 2.0 - _theta1;
+
+  // Change the third angle to be measured from the +x axis
+  _theta2 = _theta2 - libMesh::pi / 2.0;
+
+  // Set up domain bounds with mesh tools
+  for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
+  {
+    _bottom_left(i) = _mesh.getMinInDimension(i);
+    _top_right(i) = _mesh.getMaxInDimension(i);
+  }
+  _range = _top_right - _bottom_left;
+
+  // Only compute the tangent once for computational efficiency
+  _tan_theta1 = std::tan(_theta1);
+  _tan_theta2 = std::tan(_theta2);
+}
+
+Real
+TricrystalTripleJunctionIC::value(const Point & p)
+{
+  /*
+   * This does all the work to create a triple junction that looks like the letter Y
+   */
+  Real dist_left; // The distance from the point to the line specified by _theta1
+  Real dist_right; // The distance from the point to the line specified by _theta2
+
+  // Check if the point is above or below the left-most line
+  // Function to use is y = max_y/2.0 + (x - max_x/2.0) * std::tan(libMesh::pi/2.0 - _theta1)
```

Comment:
Demonstrating my ignorance a bit here, does the mesh not start at ```(0,0,0)``` and end at some ```max_value```?